Cockpit 247 and Cockpit Machines 246 (#424)
cockpit-project/cockpit-project.github.io: _posts/2021-06-23-cockpit-247.md

---
title: Cockpit 247
author: kkoukiou
date: 2021-06-23
tags: cockpit linux
slug: cockpit-247
category: release
summary: Cockpit with Metrics and Virtual Machines improvements
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

> Here are the release notes from Cockpit version 247 and Cockpit Machines version 246.

Reviewer: Any reason not to skip a version with machines, in order to keep in sync?

Author: There is not any reason to keep in sync actually, these are separate projects. Podman is in release 31 for example.

Reviewer 2: All the more reason to add the various versions to the about dialog... https://github.com/cockpit-project/cockpit/issues/15962

> ## Metrics: Grafana client setup

Reviewer: please add some verb here, like "enable Grafana..." or so, or phrase it rather like "new support for ...".  the heading should describe what changed.

The "Metrics settings" button introduced in [Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html)
has a new switch to set up exporting metrics to the network through the [PCP pmproxy service](https://linux.die.net/man/1/pmproxy)
and the [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html):

![metrics settings](/images/metrics-settings-2.png)

If the machine uses firewalld, you can allow pmproxy in a particular zone afterwards:

![metrics firewall zone selector](/images/metrics-fw-sel.png)

After that, a remote [Grafana](https://grafana.com/) instance can read the machine's metrics through the
[PCP Grafana plugin](https://grafana-pcp.readthedocs.io/), as described on the [PCP metrics feature page](https://cockpit-project.org/guide/latest/feature-pcp.html).

> ## Machines: Share host files with the guest

Reviewer: I wonder if it's worth mentioning "virtio-fs" directly in the title.  The people interested in this feature may already know what it's called?

Reviewer 2: Something like this?

```suggestion
## Machines: Share host files with the guest using virtio-fs
```

Author: So they just got back to me again:

kkoukiou: groug dgilbert Lowercase "virtiofs" is the most recent spelling. Before it was virtio-fs but the Linux file system name used on the mount command-line is "virtiofs", so we recently agreed to try to use that consistently.
It doesn't help that we changed the spelling :-)
But the hope is that people won't try mount -t virtio-fs and receive an error :)

So I will adjust accordingly.

The user can now share a host directory tree with a guest. This is internally using [Virtio-fs](https://virtio-fs.gitlab.io/),
which is designed to offer local file system semantics and performance.
Virtio-fs is available in mainline since Linux 5.4, QEMU 5.0, libvirt 6.2 and Web console will not expose this feature in the UI if the requirements are not met.

![shared directories card](/images/machines-shared-directories-card.png)

![machines new shared directory dialog](/images/machines-new-shared-directory-dialog.png)

> ## Machines: list USB, PCI and SCSI host devices that are passed through to the guest VM

Reviewer: That's a bit too long.  Maybe "Machines: add list of pass-through devices"?

Author: Can it be 'Show list of ... '?

![machines host devices card](/images/machines-host-devices-card.png)

### Try it out

Cockpit 247 and Cockpit Machines 246 are available now:

 * [For your Linux system](https://cockpit-project.org/running.html)

>  * [Cockpit Source Tarball]()

Reviewer: This stuff, of course...

---
title: Cockpit 247
author: kkoukiou
date: 2021-06-23
tags: cockpit linux
slug: cockpit-247
category: release
summary: Cockpit with Metrics and Virtual Machines improvements
---

Cockpit is the [modern Linux admin interface](https://cockpit-project.org/).  We release regularly.

Here are the release notes from Cockpit version 247 and Cockpit Machines version 246.

## Metrics: Enable Grafana client setup

The "Metrics settings" button introduced in [Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html)
has a new switch to set up exporting metrics to the network through the [PCP pmproxy service](https://linux.die.net/man/1/pmproxy)

> and the [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html):

Author: You can't 'enable' a database though

> The "Metrics settings" button introduced in [Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html)
> has a new switch to set up exporting metrics to the network through the [PCP pmproxy service](https://linux.die.net/man/1/pmproxy)
> and the [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html):

Reviewer 2: Whoops.

```suggestion
"Metrics settings", [introduced in Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html), has a new option to share Performance Co-Pilot data over the network. Toggling "export to network" on enables the [pmproxy service](https://linux.die.net/man/1/pmproxy) using a [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html).
```

> The "Metrics settings" button introduced in [Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html)
> has a new switch to set up exporting metrics to the network through the [PCP pmproxy service](https://linux.die.net/man/1/pmproxy)
> and the [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html):

Reviewer 2: Is this closer?

```suggestion
"Metrics settings", [introduced in Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html), has a new option to share Performance Co-Pilot data over the network. Toggling "export to network" on enables a  [Redis](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html)-backed [pmproxy service](https://linux.die.net/man/1/pmproxy) and adjusts appropriate firewall settings.
```

> The "Metrics settings" button introduced in [Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html)
> has a new switch to set up exporting metrics to the network through the [PCP pmproxy service](https://linux.die.net/man/1/pmproxy)
> and the [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html):

Reviewer 2: Using "data source" now:

```suggestion
"Metrics settings", [introduced in Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html), has a new option to share Performance Co-Pilot data over the network. Toggling "export to network" on enables a  [Redis](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html) data source with a [pmproxy service](https://linux.die.net/man/1/pmproxy) and adjusts appropriate firewall settings.
```

> and the [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html):

Author: That's beeter. there is a double space before [Redis] and also I am not sure about the 'appropriate firewall settings'

I would rather write, adjust firewall settings as needed.  But I would leave it up to you ans @allisonkarlitskaya

> The "Metrics settings" button introduced in [Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html)
> has a new switch to set up exporting metrics to the network through the [PCP pmproxy service](https://linux.die.net/man/1/pmproxy)
> and the [Redis database](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html):

Reviewer 2: Sure, that's fine.

```suggestion
"Metrics settings", [introduced in Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html), has a new option to share Performance Co-Pilot data over the network. Toggling "export to network" on enables a [Redis](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html) data source with a [pmproxy service](https://linux.die.net/man/1/pmproxy) and adjusts firewall settings as needed.
```

> "Metrics settings", [introduced in Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html), has a new option to share Performance Co-Pilot data over the network. Toggling "export to network" on enables a [Redis](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html) data source with a [pmproxy service](https://linux.die.net/man/1/pmproxy) and adjusts firewall settings as needed.

Reviewer 3: I find this a bit confusing now. "Toggling <blah> on enables.." is hard to read for me as non-native speaker at least. "a" pmproxy service is also confusing, there is just one. Finally, the metrics settings dialog does *not* adjust firewall settings, this is done by the following paragraph.

```suggestion
"Metrics settings", [introduced in Cockpit 245](https://cockpit-project.org/blog/cockpit-245.html), has a new option to share Performance Co-Pilot data over the network. Enabling "export to network" enables a [Redis](https://grafana-pcp.readthedocs.io/en/latest/datasources/redis.html) data source with the [pmproxy service](https://linux.die.net/man/1/pmproxy).
```

![metrics settings](/images/metrics-settings-2.png)

If the machine uses firewalld, you can allow pmproxy in a particular zone afterwards:

![metrics firewall zone selector](/images/metrics-fw-sel.png)

After that, a remote [Grafana](https://grafana.com/) instance can read the machine's metrics through the
[PCP Grafana plugin](https://grafana-pcp.readthedocs.io/), as described on the [PCP metrics feature page](https://cockpit-project.org/guide/latest/feature-pcp.html).

> ## Machines: Share host files with the guest using Virtio-fs

Reviewer 2: ```suggestion
## Machines: Share host files with the guest using Virtio-FS
```

I double-checked this and changed the capitalization as used from https://libvirt.org/kbase/virtiofs.html

> Cockpit Machines can now share a host directory tree with a guest VM.

Reviewer 3: Drop the "tree". Not a strong opinion though, feel free to just ignore/resolve.

Author: Dropped.

> The user can now share a host directory tree with a guest. This is internally using [virtiofs](https://virtiofs.gitlab.io/),
> which is designed to offer local file system semantics and performance.
> virtiofs is available in mainline since Linux 5.4, QEMU 5.0, libvirt 6.2 and Web console will not expose this feature in the UI if the requirements are not met.

Reviewer 2: ```suggestion
Cockpit Machines can now share a host directory tree with a guest VM. 

[virtio-fs](https://virtiofs.gitlab.io/) powers this sharing feature, enabling local file system semantics with performance. virtio-fs has been available in mainline Linux 5.4 (November 2019), QEMU 5.0 (April 2020), and libvirt 6.2 (April 2020). Sharing host files is available only when these versions are met.
```

Author: 'enabling local file system semantics with performance'  <- this is different meaning than the original sentence.

Author: also I would avoid including the dates because different distros instroduce these package version at different points in time.

Reviewer 2: > also I would avoid including the dates because different distros instroduce these package version at different points in time

Hmm, right. It's the release dates of each to give context that it isn't a new thing.

```suggestion
Cockpit Machines can now share a host directory tree with a guest VM. 

[virtio-fs](https://virtiofs.gitlab.io/) powers this sharing feature, with local file system semantics and performance. virtio-fs has been available in mainline Linux 5.4, QEMU 5.0, and libvirt 6.2. Sharing host files is available only when these versions are met.
```

Author: we agreed to use virtiofs - after the developers contacted me second time about the wording.

Author: Also instead of 'with local... ' can we still use 'offering local...'

Since it's not clear the the underlying tool is implemented in that way, and not cockpit itself.

Reviewer 2: ```suggestion
Cockpit Machines can now share a host directory tree with a guest VM.

[virtiofs](https://virtiofs.gitlab.io/) powers this sharing feature, offering local file system semantics and performance. virtiofs has been available in mainline Linux 5.4, QEMU 5.0, and libvirt 6.2. Sharing host files is available only when these versions are met.
```

> virtiofs is available in mainline since Linux 5.4, QEMU 5.0, libvirt 6.2 and Web console will not expose this feature in the UI if the requirements are not met.

Author: That was rewritten by garrett.

![shared directories card](/images/machines-shared-directories-card.png)

![machines new shared directory dialog](/images/machines-new-shared-directory-dialog.png)

## Machines: Show list of pass-through devices

> Virtual Machines details now show the list of USB, PCI and SCSI host devices that are passed through to the guest VM.

Reviewer: "The virtual machines details page now shows..."

Author: Included.

![machines host devices card](/images/machines-host-devices-card.png)

### Try it out

Cockpit 247 and Cockpit Machines 246 are available now:

 * [For your Linux system](https://cockpit-project.org/running.html)
 * [Cockpit Source Tarball]()
 * [Cockpit Fedora 34]()
 * [Cockpit Fedora 33]()
 * [cockpit-machines Source Tarball]()
 * [cockpit-machines Fedora 34]()

>  * [cockpit-machines Fedora 33]()

Reviewer 2: ```suggestion
 * [cockpit-machines Fedora 33]()
 
 *[VM]: Virtual Machine
 *[PCP]: Performance Co-Pilot
```

Added some terms.
